src/test/storage_tests.cpp

```diff
@@ -465,4 +466,51 @@ BOOST_AUTO_TEST_CASE(LowerBoundTest)
     }
 }
 
+BOOST_AUTO_TEST_CASE(CreateFuturesMultiIndexTest)
+{
+    const CFuturesUserHeightPrefixKey key[] = { {100u, {0}, 10u}, {101u, {1}, 11u}, {102u, {2}, 12u}, {103u, {3}, 13u}, {104u, {4}, 14u} };
+    CFuturesUserValue future[] = { {{{0u}, 1000}, 50u}, {{{1u}, 1001}, 51u}, {{{2u}, 1002}, 52u}, {{{3u}, 1003}, 53u}, {{{4u}, 1004}, 54u} };
+
+    // Store future swap key value pairs CFuturesUserHeightPrefixKey
+    for(int i = 0; i < 4; ++i) {
+        BOOST_CHECK(pcustomcsview->WriteBy<CAccountsView::ByFutureSwapHeightKey>(key[i], future[i]));
+    }
+
+    // Check the db
+    for(int i = 0; i < 4; ++i) {
+        ResVal<CFuturesUserValue> result = pcustomcsview->GetFuturesUserValues(key[i]);
+        BOOST_CHECK(result);
+        BOOST_CHECK_EQUAL((*result).source, future[i].source);
+        BOOST_CHECK_EQUAL((*result).destination, future[i].destination);
+    }
+
+    // Before CreateFuturesMultiIndex, No any key value pairs ByFuturesSwapKeyOwner
+    for(int i = 0; i < 4; ++i) {
+        char c;
+        const CFuturesUserOwnerPrefixKey ownerKey = {key[i].owner, key[i].height, key[i].txn};
+        BOOST_CHECK(!pcustomcsview->ReadBy<CAccountsView::ByFutureSwapOwnerKey>(ownerKey, c));
+    }
+
+    // CreateFuturesMultiIndex
+    pcustomcsview->CreateFuturesMultiIndexIfNeeded();
+
+    // After CreateFuturesMultiIndex, There should be key value pairs CFuturesUserOwnerPrefixKey
+    for(int i = 0; i < 4; ++i) {
+        char c;
+        const CFuturesUserOwnerPrefixKey ownerKey = {key[i].owner, key[i].height, key[i].txn};
+        BOOST_CHECK(pcustomcsview->ReadBy<CAccountsView::ByFutureSwapOwnerKey>(ownerKey, c));
```

Comment:
@prasannavl @Bushstar Adding a `Flush()` at the end of  `CreateFuturesMultiIndexIfNeeded` will cause this line to fail.

Reply:
The Flush() endpoint is https://github.com/DeFiCh/ain/blob/master/src/flushablestorage.h#L300